salt/files/master.d/f_defaults.conf

```diff
@@ -746,14 +746,21 @@ rosters:
 {%- do default_keys.append('master_tops') %}
 master_tops:
 {%- for master in cfg_master['master_tops'] -%}
-  {%- if cfg_master['master_tops'][master] is string %}
-  {{ master }}: {{ cfg_master['master_tops'][master] }}
-  {%- else %}
-  {{ master}}:
+  {%- if cfg_master['master_tops'][master] is mapping %}
+  {{ master }}:
     {%- for parameter in cfg_master['master_tops'][master] %}
     {{ parameter }}: {{ cfg_master['master_tops'][master][parameter] }}
     {%- endfor -%}
-  {%- endif -%}
+  {%- elif cfg_master['master_tops'][master] is string %}
```

Comment:
No need for a separate test for `string`. Just let the `else` handle all the scalar types.